homeassistant/components/light/osramlightify.py

```diff
@@ -197,22 +236,71 @@ def turn_off(self, **kwargs):
             _LOGGER.debug("turn_off requested transition time for light:"
                           " %s is: %s ",
                           self._name, transition)
-            self._light.set_luminance(0, transition)
+            self._luminary.set_luminance(0, transition)
         else:
             transition = 0
             _LOGGER.debug("turn_off requested transition time for light:"
                           " %s is: %s ",
                           self._name, transition)
-            self._light.set_onoff(0)
-            self._state = self._light.on()
-
+        self._luminary.set_onoff(0)
+        self._state = False
         self.schedule_update_ha_state()
 
     def update(self):
         """Synchronize state with bridge."""
         self.update_lights(no_throttle=True)
-        self._brightness = int(self._light.lum() * 2.55)
-        self._name = self._light.name()
+        self._name = self._luminary.name()
+
+
+class OsramLightifyLight(Luminary, Light):
+    """Representation of an Osram Lightify Light."""
+
+    def __init__(self, light_id, light, update_lights):
+        """Initialize the light."""
+        self._light_id = light_id
+        super().__init__(light, update_lights)
+
+    def update(self):
+        """Update status of a Light."""
+        super().update()
+        self._state = self._luminary.on()
+        self._rgb = self._luminary.rgb()
+        o_temp = self._luminary.temp()
+        if o_temp == 0:
+            self._temperature = None
+        else:
+            self._temperature = color_temperature_kelvin_to_mired(
+                self._luminary.temp()
+            )
+        self._brightness = int(self._luminary.lum() * 2.55)
+
+
+class OsramLightifyGroup(Luminary, Light):
+    """Representation of an Osram Lightify Group."""
+
+    def __init__(self, group, bridge, update_lights):
+        """Init light group."""
+        self._bridge = bridge
+        self._light = None
+        self._light_ids = []
+        super().__init__(group, update_lights)
+
+    def _get_state(self):
+        """Get state of group.
+
+        The group is on, if any of the lights in on.
+        """
+        states = []
```

Comment:
This is inefficient as you now first have to gather all results before checking if any are True.

```python
lights = self._bridge.lights()
return any(lights[light_id].on() for light_id in self._light_ids)
```